forc-pkg/src/pkg.rs

```diff
@@ -2213,26 +2330,25 @@ pub const SWAY_BIN_HASH_SUFFIX: &str = "-bin-hash";
 /// when predicates are built_package.
 pub const SWAY_BIN_ROOT_SUFFIX: &str = "-bin-root";
 
-pub fn build_package_with_options(
-    manifest: &PackageManifestFile,
-    build_options: BuildOpts,
-) -> Result<BuiltPackage> {
-    let key_debug: String = "debug".to_string();
-    let key_release: String = "release".to_string();
+/// Selects the build profile from all available build profiles in the workspace using build_opts.
+fn build_profile_from_opts(
+    build_profiles: &HashMap<String, BuildProfile>,
+    build_options: &BuildOpts,
+) -> Result<(String, BuildProfile)> {
+    let key_debug = "debug".to_string();
+    let key_release = "release".to_string();
 
     let BuildOpts {
         pkg,
         print,
-        minify,
-        binary_outfile,
-        debug_outfile,
         build_profile,
         release,
         time_phases,
         tests,
-    } = build_options;
-
+        ..
+    } = build_options.to_owned();
```

Comment:
Rather than cloning the entire `BuildOpts` here, let's just clone the fields or elements that we actually need to when we need to (if we do need to).